projects/fal/src/fal/_serialization.py

```diff
@@ -97,6 +97,38 @@ def pickle_private_attr(field: ModelPrivateAttr) -> tuple[Callable, tuple]:
     _register(ModelPrivateAttr, pickle_private_attr)
 
 
+def _patch_pydantic_model_serialization() -> None:
+    # If user has created new pydantic models in his namespace, we will try to pickle those
+    # by value, which means recreating class skeleton, which will stumble upon
+    # __pydantic_parent_namespace__ in its __dict__ and it may contain modules that happened
+    # to be imported in the namespace but are not actually used, resulting in pickling errors.
+    # Unfortunately this also means that `model_rebuid()` might not work.
+    try:
+        import pydantic
+    except ImportError:
+        return
+
+    # https://github.com/pydantic/pydantic/pull/2573
+    if not hasattr(pydantic, "__version__") or pydantic.__version__.startswith("1."):
+        return
```

Comment:
We have a similar snippet in the toolkit, but I intentionally didn't dedup it since this runs on the serverside as well and might not have pydantic at all and so a helper might hurt readability.